src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -326,7 +326,7 @@ isSameShellOrSC(const DSAInfo& candidate, const DSAInfo& current, bool& same_dsa
  * Helper method for auto_flash
  */
 static int 
-updateShellAndSC(unsigned int  boardIdx, DSAInfo& candidate, bool& reboot)
+updateShellAndSC(unsigned int  boardIdx, DSAInfo& candidate, bool& reboot, bool& warm_reboot)
 {
   reboot = false;
```

Comment:
Being that warm_reboot is a return variable, shouldn't it be initialized first to a known value?  
Something like:
`warm_reboot = false;`